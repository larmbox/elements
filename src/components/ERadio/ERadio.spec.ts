import { mount } from '@vue/test-utils';
import { expect, test } from 'vitest';
import { ERadio } from '~/index';

test('mount component', async () => {
  expect(ERadio).toBeTruthy();

  const component = mount(ERadio, {
    props: {
      id: 'id',
    },
  });

  expect(component.html()).toMatchSnapshot();
  expect(component.html()).toContain(
    'EFormControl ERadio ERadio--md ERadio--primary'
  );
  expect(component.html()).not.toContain('form-field');
});

test('register css variables', async () => {
  mount(ERadio);
  expect(
    document.getElementById('EThemeProvider--global')!.innerText
  ).toContain('--ERadio');
});

test('inherit attributes', async () => {
  const component = mount(ERadio, {
    props: { notaprop: '1', variant: 'primary' },
  });
  expect(component.html()).toContain('notaprop="1"');
  expect(component.html()).not.toContain('variant="primary"');
});
