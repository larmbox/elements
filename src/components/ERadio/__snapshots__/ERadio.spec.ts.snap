// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"EFormControl ERadio ERadio--md ERadio--primary\\">
  <div class=\\"ERadio-inner\\">
    <div><input id=\\"id\\" type=\\"radio\\" />
      <div class=\\"ERadio-box\\">
        <div class=\\"ERadio-inner-box\\"></div>
      </div>
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
</div>"
`;
